geoips/interfaces/base.py

```diff
@@ -460,8 +412,59 @@ def get_plugin(self, name):
         ProductsInterface which uses a tuple containing 'source_name' and
         'name'.
         """
+        if not self._unvalidated_plugins:
+            raise PluginRegistryError(
+                "Plugin registry not found, please run 'create_plugin_registry'"
+            )
         try:
-            validated = self.validator.validate(self._unvalidated_plugins[name])
+            if isinstance(name, tuple):
+                # These are stored in the yaml as str(name),
+                # ie "('viirs', 'Infrared')"
+                plugin = yaml.safe_load(
+                    open(
+                        self._unvalidated_plugins[self.name][name[0]][name[1]][
+                            "abspath"
+                        ],
+                        "r",
+                    )
+                )
+                plugin_found = False
+                for product in plugin["spec"]["products"]:
+                    if (
+                        product["name"] == name[1]
+                        and name[0] in product["source_names"]
+                    ):
+                        plugin_found = True
+                        plugin = product
+                        break
+                if not plugin_found:
+                    raise PluginError(
+                        "There is no plugin that has " + name[1] + " included in it."
+                    )
+                plugin["interface"] = "products"
+                plugin["abspath"] = self._unvalidated_plugins[self.name][name[0]][
+                    name[1]
+                ]["abspath"]
+                plugin["relpath"] = self._unvalidated_plugins[self.name][name[0]][
+                    name[1]
+                ]["relpath"]
+                plugin["package"] = self._unvalidated_plugins[self.name][name[0]][
+                    name[1]
+                ]["package"]
+            else:
+                plugin = yaml.safe_load(
+                    open(self._unvalidated_plugins[self.name][name]["abspath"], "r")
+                )
+                plugin["abspath"] = self._unvalidated_plugins[self.name][name][
+                    "abspath"
+                ]
+                plugin["relpath"] = self._unvalidated_plugins[self.name][name][
+                    "relpath"
+                ]
+                plugin["package"] = self._unvalidated_plugins[self.name][name][
+                    "package"
+                ]
+            validated = self.validator.validate(plugin)
```

Comment:
Can this section be simplified by setting `unval_plugin = self._unvalidated_plugins[self.name][name]` then using that?

Reply:
No it can't. We have to load in the appropriate plugin, then add the appropriate information (package, relpath, abspath), to that yaml dictionary. If you can think of another method though, I'm all ears. This relates to #383.

Reply:
I think that #389 will help with this. We'll see. I'm going to close this for now.